components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -146,10 +175,18 @@ export class TeamDBImpl implements TeamDB {
             );
         }
 
+        let slug = slugify(name, { lower: true });
         const teamRepo = await this.getTeamRepo();
+
+        const existingTeam = await teamRepo.findOne({ slug, deleted: false, markedDeleted: false });
+        if (!!existingTeam) {
+            slug = slug + "-" + randomBytes(4).toString("hex");
+        }
+
         const team: Team = {
             id: uuidv4(),
             name,
+            slug,
             creationTime: new Date().toISOString(),
         };
         await teamRepo.save(team);
```

Comment:
This can race and allow duplicate slugs. Can we push the check for existing into a single `create` method which does this in a transaction?

Reply:
Good point. Using a db transaction should be feasible. ⏳

Reply:
done